### ISSUE TITLE
Process cannot exit gracefully

### ISSUE THREAD

**Issue by searls** [jersearls/table-name-converter#10]
tl;dr whenever `exit` is used, it is very easy to mask other design issues, and we are getting bitten by one of them.

So, while trying to do #9, I couldn't get the test to finish a run with the driver gracefully, and that's when I noticed that both `script/driver.rb` and `lib/toolbox.rb` will call `Kernel.exit` when a user quits or exits. Because `exit` kills the entire ruby process, it means there is no opportunity to tidy things up or run assertions that the app is working.

How i propose things should work instead is that Driver starts a loop (like `while true`), and when a user reports they want to exit it, then `break` is called. That way, the method is able to exit cleanly and the app can unwind without always crashing the entire ruby process and we can write tests.

I'd expect the stack to look like this:
1. Driver.start
   1. Loop start
      1. Start browser loop
         1. Do browser commands
         2. Indicate stop browsing
      2. Break out of browser loop
   2. Indicate Exit app
   3. Break out of Driver loop
2. Driver.start exits

This would be a pretty easy fix, but there is a design problem that was introduced by having used `exit` and calling `home` in `lib/toolbox.rb`. It means that the stack of functions when you're, say, browsing is actually sorta like this:
1. Driver.start
   1. Enter browser
      1. Do browser commands
      2. Indicate stop browsing
      3. Call Driver.start again 
         1. Indicate Exit app
         2. Crash process with `exit`

If I simply remove the `exit` call from the Driver (or Toolbox), then you'll just be punted back up into the browser.  Before we can get a test to run, we need to refactor the app to have a simpler stack like the first outline above so it can exit cleanly

**Comment by searls** [comment]
:sunny:  :sparkles:
